.pre-commit-config.yaml

```diff
@@ -1,71 +1,81 @@
-#    - --safe
-#    language_version: python3.8
-#- repo: https://github.com/asottile/blacken-docs
-#  rev: v1.8.0
-#  hooks:
-#  - id: blacken-docs
-#    additional_dependencies:
-#    - black==19.10b0
-#    language_version: python3.8
-
-#- repo: https://github.com/asottile/add-trailing-comma
-#  rev: v2.0.1
-#  hooks:
-#  - id: add-trailing-comma
-#- repo: https://github.com/pre-commit/pygrep-hooks
-#  rev: v1.6.0
-#  hooks:
-#  - id: rst-backticks
-#- repo: https://github.com/asottile/setup-cfg-fmt
-#  rev: v1.11.0
-#  hooks:
-#  - id: setup-cfg-fmt
-#    args:
-#    - --min-py3-version
-#    - '3.7'
-#- repo: https://gitlab.com/pycqa/flake8
-#  rev: 3.8.3
-#  hooks:
-#  - id: flake8
-#    additional_dependencies:
-#    - flake8-bugbear == 20.1.2
-#    language_version: python3.8
+      - id: check-ast
+      - id: check-builtin-literals
+      - id: check-docstring-first
+      - id: check-merge-conflict
+      - id: check-yaml
+      - id: check-toml
+      - id: debug-statements
+#      - id: end-of-file-fixer
+#        exclude: 'tests/test-data'
+#      - id: trailing-whitespace
+#        exclude: 'tests/test-data'
+#  - repo: https://github.com/asottile/add-trailing-comma
+#    rev: v2.4.0
+#    hooks:
+#      - id: add-trailing-comma
+#        args: [--py36-plus]
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v3.3.1
+    hooks:
+      - id: pyupgrade
+        args: ["--py38-plus"]
+#  - repo: https://github.com/PyCQA/isort
+#    rev: v5.11.3
+#    hooks:
+#      - id: isort
+  - repo: https://github.com/psf/black
+    rev: 22.12.0
+    hooks:
+      - id: black
+        args: [--safe]
+  - repo: https://github.com/asottile/blacken-docs
+    rev: v1.12.1
+    hooks:
+      - id: blacken-docs
+        additional_dependencies: [black==22.12]
+  - repo: https://github.com/pre-commit/pygrep-hooks
+    rev: v1.9.0
+    hooks:
+      - id: rst-backticks
+  - repo: https://github.com/tox-dev/tox-ini-fmt
+    rev: "0.5.2"
+    hooks:
+      - id: tox-ini-fmt
+        args: ["-p", "fix_lint"]
+  - repo: https://github.com/tox-dev/pyproject-fmt
+    rev: "0.1.0"
+    hooks:
+      - id: pyproject-fmt
+#  - repo: https://github.com/PyCQA/flake8
+#    rev: 6.0.0
+#    hooks:
+#      - id: flake8
+#        additional_dependencies:
+#          - flake8-bugbear==22.12.6
+#          - flake8-comprehensions==3.10.1
+#          - flake8-pytest-style==1.6
+#          - flake8-spellcheck==0.28
+#          - flake8-unused-arguments==0.0.12
+#          - flake8-noqa==1.3
+#          - pep8-naming==0.13.3
+#          - flake8-pyproject==1.2.2
+#  - repo: https://github.com/pre-commit/mirrors-prettier
+#    rev: "v2.7.1"
+#    hooks:
+#      - id: prettier
+#        additional_dependencies:
+#          - prettier@2.7.1
+#          - "@prettier/plugin-xml@2.2"
+#        args: ["--print-width=120", "--prose-wrap=always"]
+#  - repo: https://github.com/igorshubovych/markdownlint-cli
+#    rev: v0.32.2
+#    hooks:
+#      - id: markdownlint
```

Comment:
This one is pretty cool for markdown validation, but too many false positives due to unknown words (like... k-mer =]). Need to figure how to configure it properly.